session/sql_store.go

```diff
@@ -745,9 +745,12 @@ func unmarshalSession(ctx context.Context, db SQLQueries,
 
 	var macRecipe *MacaroonRecipe
 	if perms != nil || caveats != nil {
-		macRecipe = &MacaroonRecipe{
-			Permissions: unmarshalMacPerms(perms),
-			Caveats:     unmarshalMacCaveats(caveats),
+		macRecipe = &MacaroonRecipe{}
+		if perms != nil {
+			macRecipe.Permissions = unmarshalMacPerms(perms)
+		}
+		if caveats != nil {
+			macRecipe.Caveats = unmarshalMacCaveats(caveats)
```

Comment:
Let me know if you think we should rather fix the kvdb store implementation rather than the sql one. But as we've previously opted for not updating the store behaviour, and make the sql one mimic the behaviour until we've completed the migration, I opted for this version here.

Reply:
Alternatively, I could also override this in newly added `overrideMacaroonRecipe` function before the migration deep equals check. But I kept this as is, as I do think it's correct behaviour that we don't set the `Permissions` or `Caveats` and keep them at `nil`, if no value actually exists for them.

Reply:
Please see https://github.com/lightninglabs/lightning-terminal/pull/1051#issuecomment-2934569218